flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionVertexStopTest.java

```diff
@@ -0,0 +1,141 @@
+import org.apache.flink.runtime.messages.Messages;
+import org.apache.flink.runtime.messages.TaskMessages;
+import org.apache.flink.runtime.messages.TaskMessages.TaskOperationResult;
+import org.apache.flink.runtime.testingUtils.TestingUtils;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import akka.actor.ActorSystem;
+import akka.testkit.JavaTestKit;
+import scala.concurrent.ExecutionContext;
+import static org.apache.flink.runtime.executiongraph.ExecutionGraphTestUtils.setVertexState;
+import static org.apache.flink.runtime.executiongraph.ExecutionGraphTestUtils.getExecutionVertex;
+import static org.apache.flink.runtime.executiongraph.ExecutionGraphTestUtils.getInstance;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(ExecutionVertex.class)
+public class ExecutionVertexStopTest {
+
+	@Test
+	public void testStop() throws Exception {
+		final JobVertexID jid = new JobVertexID();
+		final ExecutionJobVertex ejv = getExecutionVertex(jid);
+
+		Execution executionMock = mock(Execution.class);
+		whenNew(Execution.class).withAnyArguments().thenReturn(executionMock);
+
+		final ExecutionVertex vertex = new ExecutionVertex(ejv, 0, new IntermediateResult[0],
+				AkkaUtils.getDefaultTimeout());
+
+		vertex.stop();
+
+		verify(executionMock).stop();
+	}
+
+	private static ActorSystem system;
+
+	@BeforeClass
+	public static void setup(){
+	}
+
+	@AfterClass
+	public static void teardown(){
+		if(system != null) {
+			JavaTestKit.shutdownActorSystem(system);
+			system = null;
+		}
+	}
+
+	static boolean receivedStopSignal;
+
+	@Test
+	public void testStopRpc() {
+		try {
+			final JobVertexID jid = new JobVertexID();
+			final ExecutionJobVertex ejv = getExecutionVertex(jid);
+
+			final ExecutionVertex vertex = new ExecutionVertex(ejv, 0, new IntermediateResult[0],
+					AkkaUtils.getDefaultTimeout());
+			final ExecutionAttemptID execId = vertex.getCurrentExecutionAttempt().getAttemptId();
+
+			setVertexState(vertex, ExecutionState.SCHEDULED);
+			assertEquals(ExecutionState.SCHEDULED, vertex.getExecutionState());
+
+			final ActorGateway gateway = new StopSequenceInstanceGateway(
+					TestingUtils.defaultExecutionContext(), new TaskOperationResult(execId, true));
+
+			Instance instance = getInstance(gateway);
+			SimpleSlot slot = instance.allocateSimpleSlot(new JobID());
+
+			vertex.deployToSlot(slot);
+
+			receivedStopSignal = false;
+			vertex.stop();
+			assertTrue(receivedStopSignal);
+		}
+		catch(Exception e) {
+			e.printStackTrace();
+			fail(e.getMessage());
```

Comment:
Why do you print the stack trace and fail the test with an `AssertionError` where it would also fail with the actual exception?